### PR TITLE
feat(function): Add methods CustomCollectionAction, CustomCollectionF…

### PR DESCRIPTION
**Pull request by bastienlemaitre**
Add methods CustomCollectionAction, CustomCollectionFunction and support for Apply.

I've tried to create others tests but i dont know if your api contains CollectionActions and CollectionFunctions.

It match the Microsoft's tutorial about Functions & Actions for OData V4:
https://blogs.msdn.microsoft.com/odatateam/2014/12/08/tutorial-sample-functions-actions-in-web-api-v2-2-for-odata-v4-0-type-scenario/

### CODE REVIEW

File: src/angularODataQuery.ts [hunk 1]
```diff
@@ -9,10 +9,12 @@ import { ODataPagedResult } from './angularODataPagedResult';
 import { IODataResponseModel } from './angularODataResponseModel';
 
 export class ODataQuery<T> extends ODataOperation<T> {
+
     private _filter: string[] = [];
     private _top: number;
     private _skip: number;
     private _orderBy: string[] = [];
+    private _apply: string[];
```
Comment (StefH) [inline src/angularODataQuery.ts]: Init an empty array here, else the unit-tests fail.

``` javascript
private _apply: string[] = [];
```

**Review by StefH — CHANGES_REQUESTED** [review]
I did add a code-review comment which solves some/all failures in the unit-tests.

Can you take a look ?

File: test/angularODataQuery.spec.ts [hunk 2]
```diff
@@ -200,4 +200,15 @@ describe('ODataQuery', () => {
...
+
+    it('Apply(string)', inject([HttpClient, ODataConfiguration], (http: HttpClient, config: ODataConfiguration) => {
+        // Assign
+        const test = new ODataQueryMock('Employees', config, http);
+
+        // Act
+        test.Apply('groupby((LastName))');
+
+        // Assert
+        assert.equal(test['_apply'], ['groupby((LastName))']);
```
Comment (StefH) [inline test/angularODataQuery.spec.ts]: use `deepEqual`

**Review by StefH — CHANGES_REQUESTED** [review]

**Comment by StefH** [comment]
I've added some code-review comments, can you take a look?

**Comment by bastienlemaitre** [comment]
Fixed

**Comment by bastienlemaitre** [comment]
Can you add methods on your api like this tutorial ? After I will add more units tests.
https://blogs.msdn.microsoft.com/odatateam/2014/12/08/tutorial-sample-functions-actions-in-web-api-v2-2-for-odata-v4-0-type-scenario/

**Review by StefH — CHANGES_REQUESTED** [review]
You have replaced/removed the tests for
Custom Function 1
Custom Function 2
Custom Function with parameters

Please add these again.

**Comment by StefH** [comment]
You have replaced/removed the tests for
Custom Function 1
Custom Function 2
Custom Function with parameters

Please add these again.

**Comment by bastienlemaitre** [comment]
I will do this.

**Comment by bastienlemaitre** [comment]
Fixed

**Comment by StefH** [comment]
One more request: this line is not tested:
https://codecov.io/gh/StefH/angular-odata-es5/compare/c0cc4519309ff239edd29742a643c8a21f0d4617...c29f052f554dc6163aeac309aa9f7fb8b3985f82/src/src/angularODataQuery.ts?before=src/angularODataQuery.ts#L124

A solution for this is adding a `it('Exec' . . . ` test in the file `odataAngularQuery.spec.ts` which tests this new Apply code

**Comment by bastienlemaitre** [comment]
Fixed

**Comment by bastienlemaitre** [comment]
Can you merge it ?

**Comment by StefH** [comment]
I'll merge it in some time.

**Comment by bastienlemaitre** [comment]
Sorry to be insistent but I absolutely need it for my work. No chance to merge?

**Review by StefH — APPROVED** [review]

**Comment by StefH** [comment]
Version 0.2.8 is added to npmjs.

**Comment by bastienlemaitre** [comment]
Thank you !

**Pull request merged**